pkg/fleets/controller_test.go

```diff
@@ -1118,11 +1109,6 @@ func TestFeatureRollingUpdateOnReady(t *testing.T) {
 func TestControllerRollingUpdateDeployment(t *testing.T) {
```

Comment:
I will rerun it locally and check the problem

Reply:
@roberthbailey I can reproduce the test failures in my env. Seems change in controller.go is straightforward.  I didn't make other changes. Not sure why this test failed.   Do you have any clues?

Reply:
Given that these tests were aimed at when the feature was off `assert.NoError(t, utilruntime.ParseFeatures(string(utilruntime.FeatureRollingUpdateOnReady)+"=false"))`, it's expected they would fail now it's enabled always.

So they would either need to be updated to the correct values, or removed entirely.

Reply:
I took a quick look at these tests and it isn't immediately obvious to me what we are checking for. What test coverage are we losing if we delete the entire set of tests? Or just delete the 3 that are failing?

Reply:
This tests various gameserver set and rolling update deployment scenarios, and if they go beyond max surge, remove items as appropriate, etc.

Basically does `rollingUpdateDeployment` do as it should.

I think it's worth converting this over before moving to stable - also to ensure an extra layer that it is doing as expected.

Reply:
That makes sense to me. @Jeffwan - we have about 1 more day to try and get this in before the 1.18 release, otherwise we can ship it with 1.19.

Reply:
em. Seem it's better to change to the appropriate values rather than removing them? Let me check algorithm today and see if I can make the changes.

Reply:
I think I figure out the problem. I manually change to use `c.rollingUpdateRestBeforeFixOnReady()` they all pass. They do work for case `utilruntime.FeatureRollingUpdateOnReady=false`. 

https://github.com/googleforgames/agones/blob/aefac73bad27949e745e6354f709c70f1646470b/pkg/fleets/controller.go#L605-L610

Since we already use `rollingUpdateRestFixedOnReady` and remove `c.rollingUpdateRestBeforeFixOnReady`, these test need changes. 


1. I go though the logic and change right number in test cases "test smalled numbers of active and allocated" and "attempt to drive replicas over the max surge" 

2. I remove "statuses don't match the spec. nothing should happen", this is because update won' be called in the past `c.rollingUpdateRestBeforeFixOnReady `. However, this test case is not valid anymore because `c.rollingUpdateRestFixedOnReady ` does consider active GSS.


@roberthbailey @markmandel Please have another check.